tools/run-test.py

```diff
@@ -50,22 +68,151 @@ def run_check(runnable):
 
     return ret
 
-def download_sq_server(version):
+def download_sq_server(version, dst):
+    print('Downloading: SonarQube server (version %s)...' % version)
     sq = urllib2.urlopen('https://sonarsource.bintray.com/Distribution/'
                          'sonarqube/sonarqube-%s.zip' % version)
     dl = sq.read()
-    with open('sonarqube-%s.zip' % version, 'w') as f:
+    path = os.path.join(dst, 'sonarqube-%s.zip' % version)
+    with open(path, 'wb') as f:
         f.write(dl)
+    print('SonarQube server download completed! (version %s)' % version)
+
+def download_sq_scanner(version, dst):
+    print('Downloading: SonarQube scanner (version %s)...' % version)
+    sq = urllib2.urlopen('https://sonarsource.bintray.com/Distribution/'
+                         'sonar-scanner-cli/sonar-scanner-cli-%s-windows.zip' % version)
+    dl = sq.read()
+    path = os.path.join(dst, 'sonar-scanner-cli-%s-windows.zip' % version)
+    with open(path, 'wb') as f:
+        f.write(dl)
+    print('SonarQube scanner download completed! (version %s)' % version)
+
+def unzip(file, dst):
+    print('Unzipping files...')
+    zip_ref = zipfile.ZipFile(file, 'r')
+    zip_ref.extractall(dst)
+    zip_ref.close()
+    print('Unzip finished! (%s)' % file)
+
+def copy_all_files_from_folder(src, dst):
+    if not 'none' == src:
+        src_files = os.listdir(src)
+        for file_name in src_files:
+            full_file_name = os.path.join(src, file_name)
+            if (os.path.isfile(full_file_name)):
+                shutil.copy(full_file_name, dst)
+    else:
+        for i in range(0, 2):
+            if 0 == i:
+                language = 'core'
+            if 1 == i:
+                language = 'gui'
+            path = ['src', 'sonarqube-%s-plugin' % language, 'target', 'sourcemeter-%s-plugin-1.0.0.jar' % language]
+            path  = os.path.join(*path)
+            shutil.copy(path, dst)
+
+        for i in range(0, 5):
+            if 0 == i:
```

Comment:
This can be written in a much simpler way.
```py
languages = ['cpp', 'csharp', 'java', 'python', 'rpg']
for language in languages:
    path = ['src', 'sonarqube-analyzers', 'sourcemeter-analyzer-%s' % language, 'target', 'sourcemeter-analyzer-%s-plugin-1.0.0.jar' % language]
    path  = os.path.join(*path)
    shutil.copy(path, dst)
```